components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,126 @@
+
+Unfortunately the Kubeless functions don't have conditions so we can't check them like that.
+
+When function exceed resource quota limit, it looks like this: 
+
+```yaml
+"spec": {
+                "deployment": {
+                    "metadata": {
+                        "creationTimestamp": null
+                    },
+                    "spec": {
+                        "strategy": {},
+                        "template": {
+                            "metadata": {
+                                "creationTimestamp": null
+                            },
+                            "spec": {
+                                "containers": null
+                            }
+                        }
+                    },
+                    "status": {}
+                },
+                "horizontalPodAutoscaler": {
+                    "metadata": {
+                        "creationTimestamp": null
+                    },
+                    "spec": {
+                        "maxReplicas": 0,
+                        "scaleTargetRef": {
+                            "kind": "",
+                            "name": ""
+                        }
+                    },
+                    "status": {
+                        "conditions": null,
+                        "currentMetrics": null,
+                        "currentReplicas": 0,
+                        "desiredReplicas": 0
+                    }
+                },
+
+```
+
+## Expose a warning about exceeding resource quota.
+
+The idea is to have an information about exceeding resource quota. The ui-api-layer gives a number of pods, which can be deployed and not exceed resource quota.
+Such number is computed as:
+
+```availableByLimit = floor((resourcequota.spec.hard.limits.memory - resourcequota.status.used.limits.memory) / (limitrange.spec.default.memory))```
+
+and
+
+```availableByRequest = floor((resourcequota.spec.hard.requests.memory - resourcequota.status.used.requests.memory) / (limitrange.spec.defaultRequest.memory))```
+
+available = MIN(availableByLimit, availableByRequest) // lower value is taken
+
+The GraphQL endpoint could return such number of default pods which can be started in the namespace.
+
+
+## The simplest way
+
+The endpoint returns the resource quota values (limit and request) and used value like the command:
+```bash
+kubectl describe ns kyma-system
+```
+```bash
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+
+```
+
+The endpoint returns:
+hard
+used
+
+In case of used value is close to hard value - the UI shows an alert.
+
+## The simplest way with a percentage of usage
+
+The endpoint returns a percentage usage of resources in the given namespace.
+UI can always show actual usage for every environment and it shows warning when a percent of usage is too high.
+
+The endpoint returns:
+usage
```

Comment:
`The GraphQL endpoint returns the information about the percentage of usage.`

Reply:
Done